src/foam/dao/pg/SQLData.java

```diff
@@ -0,0 +1,144 @@
+package foam.dao.pg;
+
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+
+import java.util.*;
+import java.util.ArrayList;
+import java.util.StringJoiner;
+
+/**
+ * prepares sql info based on a FObject
+ */
+public class SQLData {
+
+  //TODO(drish): use uuids ?
+  protected Object id;
+  protected String table;
+  protected ArrayList<String> columnNames;
+  protected ArrayList<Object> values;
+
+  public SQLData(FObject obj) throws IllegalStateException, IllegalAccessException {
+
+    table = obj.getClass().getSimpleName().toLowerCase();
+    columnNames = new ArrayList<String>();
+    List<PropertyInfo> props = obj.getClassInfo().getAxioms();
+
+    // TODO(drish): throw in case where fobject has no properties ?
+    if ( props.size() <= 0 ) return;
+
+    values = new ArrayList<Object>();
+
+    for ( PropertyInfo p: props ) {
+
+      // do not include ID into columns, since its auto-incremented
+      if ( p.getName().equals("id") ) {
+        id = p.get(obj);
+        continue;
+      }
+
+      columnNames.add(p.getName());
+      values.add(p.get(obj));
+    }
+  }
+
+  /**
+   * creates a prepared statement for delete
+   * in the following format:
+   *
+   * delete from tableName where id = id
+   *
+   * @return String
+   */
+  public String createDeleteStatement() {
+    StringBuilder sql = new StringBuilder("delete from " + getTableName());
+    sql.append(" where id = ?");
+    return sql.toString();
+  }
+
+  /**
+   * creates a prepared statement for update
+   * in the following format:
+   *
+   *  update tableName set (col1, col2) = (?, ?, ?) where id = id;
+   *
+   * @return String
+   */
+  public String createUpdateStatement() {
```

Comment:
These prepared statements will never change for a particular DAO, so it's very inefficient to generate them for each statement rather than just reuse them.